### PR TITLE
Remove deprecated commons-lang3 RandomUtils class usage.

### PR DESCRIPTION
**Pull request by abhioncbr**
**Labels**
- `refactor`
- `cleanup`

**Description**
- As per the [comment](https://github.com/apache/pinot/pull/13480#discussion_r1655289078) of the [PR](https://github.com/apache/pinot/pull/13480), This PR removes the usage of commons-lang3 `RandomUtils` class.
- As per the deprecation suggestions, We have introduced the `commons-rng` dependency in Pinot.
- Added a `RandomNumberUtils` class to generate random numbers based on the commons-rng.

### CODE REVIEW

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/relocation/SegmentRelocatorTest.java [hunk 1]
```diff
@@ -126,7 +127,7 @@ public void testRebalanceTablesSequentially()
       relocator.putTableToWait("t_" + i);
     }
     for (int i = 0; i < cnt; i++) {
-      relocator.putTableToWait("t_" + RandomUtils.nextInt(0, cnt));
+      relocator.putTableToWait("t_" + RandomNumberUtils.getRandomProvider().nextInt(0, cnt));
```
Comment (Jackie-Jiang) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/relocation/SegmentRelocatorTest.java:130]: Avoid creating random number provider in a for loop

**Review by Jackie-Jiang — COMMENTED** [review]
Seems all the usages are within the tests, and this library is not that commonly used (only [71 usages](https://mvnrepository.com/artifact/org.apache.commons/commons-rng-simple)). I'd suggest just using regular java `Random()` for readability

**Comment by abhioncbr** [comment]
> Seems all the usages are within the tests, and this library is not that commonly used (only [71 usages](https://mvnrepository.com/artifact/org.apache.commons/commons-rng-simple)). I'd suggest just using regular java `Random()` for readability

I agree that this library isn't widely used and that releases are minimal, but if we are looking for performance gains, we can use it. Here are the stats related to [performance](https://commons.apache.org/proper/commons-rng/userguide/rng.html#a4._Performance). 
Based on these stats, do you think we will try to replace `java.util.Random` usage with this in our entire codebase?

**Comment by Jackie-Jiang** [comment]
We rarely use random in production code (especially in high perf code), so I'd suggest just using `Random` for now and revisit if we need high performance random generator

File: pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunctionTest.java [hunk 2]
```diff
@@ -203,9 +202,11 @@ public void setUp()
         _doubleMV2Values[i][j] = 1.0;
       }
 
+      float range = 1.0f - 0.0f;
+      Random random = new Random();
       for (int j = 0; j < VECTOR_DIM_SIZE; j++) {
-        _vector1Values[i][j] = Math.abs(RandomUtils.nextFloat(0.0f, 1.0f));
-        _vector2Values[i][j] = Math.abs(RandomUtils.nextFloat(0.0f, 1.0f));
+        _vector1Values[i][j] = Math.abs(random.nextFloat() * range + 0.0f);
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunctionTest.java:208]: Per the documentation, `Random.nextFloat()` will generate a value from 0.0 to 1.0
```suggestion
        _vector1Values[i][j] = random.nextFloat();
```

File: pinot-core/src/test/java/org/apache/pinot/queries/NoDictionaryCompressionQueriesTest.java [hunk 3]
```diff
@@ -216,11 +215,11 @@ private List<GenericRow> createTestData() {
 
     //Generate random data
     int rowLength = 1000;
-    Random random = new Random();
     String[] tempStringRows = new String[rowLength];
     Integer[] tempIntRows = new Integer[rowLength];
     Long[] tempLongRows = new Long[rowLength];
 
+    ThreadLocalRandom random = ThreadLocalRandom.current();
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/queries/NoDictionaryCompressionQueriesTest.java:222]: (minor) Use regular `Random` to be consistent with other tests

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/VectorTest.java [hunk 4]
```diff
@@ -331,7 +331,9 @@ private float[] createZeroVector(int vectorDimSize) {
 
   private float[] createRandomVector(int vectorDimSize) {
     float[] vector = new float[vectorDimSize];
-    IntStream.range(0, vectorDimSize).forEach(i -> vector[i] = RandomUtils.nextFloat(0.0f, 1.0f));
+    float range = 1.0f - 0.0f;
+    Random random = new Random();
+    IntStream.range(0, vectorDimSize).forEach(i -> vector[i] = random.nextFloat() * range + 0.0f);
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/VectorTest.java:336]:
```suggestion
    Random random = new Random();
    for (int i = 0; i < vectorDimSize; i++) {
      vector[i] = random.nextFloat();
    }
```

**Review by Jackie-Jiang — APPROVED** [review]
Thanks for cleaning this up!

**Pull request merged**